### PR TITLE
Remove .nav-item from .nav-link to make it consistent

### LINKED ISSUE #28423: Non ul or ol based .nav-link should require .nav-item class
The non `ul` or `ol` based navs e.g.

```html
<nav class="nav">
  <a class="nav-link active" href="#">Active</a>
  <a class="nav-link" href="#">Link</a>
  <a class="nav-link" href="#">Link</a>
</nav>
```

should be require `.nav-item` class:

```html
<nav class="nav">
  <a class="nav-item nav-link active" href="#">Active</a>
  <a class="nav-item nav-link" href="#">Link</a>
  <a class="nav-item nav-link" href="#">Link</a>
</nav>
```

Because there are many snippets without `.nav-item` class in the docs, so it looks like it's not necessary, but the `.nav-item` class is often required. As an example, the pill style does not require `.nav-item`, but the tab style needs it (this usage isn't documented).

image

demo: https://codepen.io/anon/pen/wOJWJo

And the [example in the navbar nav](https://getbootstrap.com/docs/4.3/components/navbar/#nav) use `.nav-item` with `.nav-link`.

image

This lack of consistency caused the problem of #28117.

I strongly recommend add `.nav-item` to all `.nav-link` of the non `<ul>` or `<ol>` based nav.

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Remove the class `.nav-item` from `.nav-link` inside the non `ul` or `ol` based navs.
This makes the consistency for `.nav-item` (This class will not be required on the `.nav-link`).
`.nav-item` was only required when you use `.nav-fill` or `.nav-justified` on the `.nav`. In other cases, it was redundant. And the purpose of `.nav-item` on `.nav-link`s can be achieved via flexbox utilities as well (Mentioned in the docs also).

Closes #28423
Related #30880, #28117
(This PR covers #30912 as well)

### CODE REVIEW

**Review by mdo — APPROVED** [review]

File: site/content/docs/5.0/components/navs.md [hunk 1]
```diff
@@ -187,14 +187,14 @@ Force your `.nav`'s contents to extend the full available width one of two modif
 </ul>
 {{< /example >}}
 
-When using a `<nav>`-based navigation, be sure to include `.nav-item` on the anchors.
+When using a `<nav>`-based navigation.
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/navs.md:190]: This seems off now

Comment (rohit2sharma95) [reply site/content/docs/5.0/components/navs.md:190]: Unable to understand @XhmikosR 😊
Can you please define?

Comment (XhmikosR) [reply site/content/docs/5.0/components/navs.md:190]: The sentence seems weird with that part removed :) Maybe it should be reworded?

Comment (rohit2sharma95) [reply site/content/docs/5.0/components/navs.md:190]: Or you can even use a `<nav>`-based navigation.
@XhmikosR?

Comment (mdo) [inline site/content/docs/5.0/components/navs.md:190]:
```suggestion
When using a `<nav>`-based navigation, you can safely omit `.nav-item` as only `.nav-link` is required for styling `<a>` elements.
```

**Pull request merged**